### ISSUE TITLE
Constructor documentation needs to move from __init__ to class docstring

### ISSUE THREAD

**Issue by pushfoo** [pythonarcade/arcade#1005]
### Actual behavior:

**tl;dr The current Sphinx configuration is omitting all constructor docstrings, including their param annotations.**

An important core class that beginners _**have**_ to use is affected: `arcade.SpriteList`. It is important that we include constructor argument explanations rather than just the attribute lists because constructors sometimes expose arguments that attributes do not.

`SpriteList`'s constructor docstring contains thorough explanations for each argument:
https://github.com/pythonarcade/arcade/blob/2ad27893c66efec62e71a3316b177b0fd25f0dfe/arcade/sprite_list/sprite_list.py#L48-L67

However, the current documentation build does not explain any of it:
![image](https://user-images.githubusercontent.com/36696816/137123144-76a12742-0e2c-43fe-a19c-ea01aeaa69e9.png)

Some other classes affected by this include arcade.TileMap and arcade.Texture. I'm sure how many others there are.

### Expected behavior:

Constructor documentation is output:
![image](https://user-images.githubusercontent.com/36696816/137126626-c6a95e68-ccf2-4084-b4cc-58f4abaa8726.png)

### Steps to reproduce/example code:
build arcade normally with `make docs`

###  ~~One way to fix it~~

~~Make sphinx use search both constructor docstrings. This seems like a good idea because:~~
~~1. Classes seem to document their options in only one place, not both~~
~~2. Sphinx has a simple config option to enable this [autoclass_content](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autoclass_content)~~
~~3. Making the config change will reveal deeper information to curious beginners immediately~~

~~Adding `autoclass_content = 'both'` in `doc\conf.py` appears to generate doc properly for the classes that currently don't print docstrings. We may need to do some cleanup work around this.~~

**Comment by pushfoo** [comment]
Discord discussion with pcraven clarified that the project standard is putting constructor arguments in the class-level docstring for objects. This now a pure documentation cleanup ticket rather than a bug fix ticket.

The listed classes should have their constructor param annotations moved to the class level. I can take care of this later.

**Comment by einarf** [comment]
I fixed the SpriteList docs now. Probably more clases with this issue. It's really something we should prioritize fixing. Even my cousin have commented on that 😄

**Comment by pushfoo** [comment]
Working on it. I also started looking into quick and dirty automated detection.
